tests/test_mixture.py

```diff
@@ -715,7 +716,8 @@ def test_mixture_with_DiracDelta():
     assert m_vv in logp_res
 
 
-def test_switch_mixture():
+@pytest.mark.parametrize("op", [at.switch, ifelse])
+def test_switch_ifelse_mixture(op):
```

Comment:
Now that you have it parametrized, let's add parameters for a new `size` parameter on each `RandomVariable`.  In other words, we should make sure that the dimensions are handled correctly.

Reply:
> The condition for `ifelse` needs to be a [scalar](https://github.com/aesara-devs/aesara/blob/main/aesara/ifelse.py#L162). Should I add the `size` parameter for only `switch` tests?

Yes, buy the branch values don't need to be scalar.

Reply:
On a related note, would the condition for a switch need to be a scalar too? [Here], I enforce mixing indices to be a `ScalarVariable` but, when using vector indices and evaluating the `(factorized_)joint_logprob`, I get directed to [this error](https://github.com/aesara-devs/aesara/blob/main/aesara/tensor/basic.py#L595-L597) that the inputs (here indices) need to be scalar too. To make the question more clear, here is the example that I was working with:

```python
srng = at.random.RandomStream(seed=2320)

I_rv = srng.bernoulli(0.5, name="I", size=(5,))
X_rv = srng.normal(-10, 0.1, name="X", size=(5,))
Y_rv = srng.normal(10, 0.1, name="Y", size=(5,))

Z_rv = at.switch(I_rv, X_rv, Y_rv)
Z_rv.name = "Z"

z_vv = Z_rv.clone()
i_vv = I_rv.clone()

logp_parts = factorized_joint_logprob({Z_rv: z_vv, I_rv: i_vv}, sum=False)
```

<details><summary>Here is the error traceback</summary>

```python
ERROR (aesara.graph.rewriting.basic): SequentialGraphRewriter apply <aesara.graph.rewriting.basic.EquilibriumGraphRewriter object at 0x166c30eb0>
ERROR (aesara.graph.rewriting.basic): Traceback:
ERROR (aesara.graph.rewriting.basic): Traceback (most recent call last):
  File "/Users/larryshamalama/Documents/GitHub/aesara/aesara/graph/rewriting/basic.py", line 302, in apply
    sub_prof = rewriter.apply(fgraph)
  File "/Users/larryshamalama/Documents/GitHub/aesara/aesara/graph/rewriting/basic.py", line 2447, in apply
    node_rewriter_change = self.process_node(
  File "/Users/larryshamalama/Documents/GitHub/aesara/aesara/graph/rewriting/basic.py", line 1933, in process_node
    replacements = node_rewriter.transform(fgraph, node)
  File "/Users/larryshamalama/Documents/GitHub/aesara/aesara/graph/rewriting/basic.py", line 1092, in transform
    return self.fn(fgraph, node)
  File "/Users/larryshamalama/Documents/GitHub/aeppl/aeppl/mixture.py", line 341, in switch_ifelse_mixture_replace
    *([NoneConst, as_nontensor_scalar(node.inputs[0])] + mixture_rvs)
  File "/Users/larryshamalama/Documents/GitHub/aesara/aesara/tensor/subtensor.py", line 677, in as_nontensor_scalar
    return aesara.tensor.scalar_from_tensor(a)
  File "/Users/larryshamalama/Documents/GitHub/aesara/aesara/tensor/basic.py", line 593, in __call__
    return type_cast(ScalarVariable, super().__call__(*args, **kwargs))
  File "/Users/larryshamalama/Documents/GitHub/aesara/aesara/graph/op.py", line 297, in __call__
    node = self.make_node(*inputs, **kwargs)
  File "/Users/larryshamalama/Documents/GitHub/aesara/aesara/tensor/basic.py", line 598, in make_node
    raise TypeError("Input must be a scalar `TensorType`")
TypeError: Input must be a scalar `TensorType`
```

Reply:
Loo

> [Here](https://github.com/aesara-devs/aeppl/blob/main/aeppl/mixture.py#L341), I enforce mixing indices to be a `ScalarVariable` but, when using vector indices and evaluating the `(factorized_)joint_logprob`, I get directed to [this error](https://github.com/aesara-devs/aesara/blob/main/aesara/tensor/basic.py#L595-L597) that the inputs (here indices) need to be scalar too.

Looks like you might need to remove that manual use of `as_nontensor_scalar`.

Reply:
I used it because it is used [here](https://github.com/aesara-devs/aesara/blob/main/aesara/tensor/subtensor.py#L705) for `Subtensor` indexing and it allows `assert equal_computations` to return true, more specifically when checking that the indices are equal.

Now, it seems to yield some restrictions for sub-graphs defined by `switch`. How should I manually remove the use of `as_nontensor_scalar` while having `equal_computations` being true when compared to a mixture sub-graph defined via a `stack`?

Reply:
I see now. `as_nontensor_scalar` worked because I only tested on cases where indices were scalars.